// @swagger/typescript-on-time-app-api/bundles/swagger-typescript-on-time-app-api.umd.js
(function (global, factory) {
    typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('@angular/core'), require('@angular/common/http')) :
    typeof define === 'function' && define.amd ? define('@swagger/typescript-on-time-app-api', ['exports', '@angular/core', '@angular/common/http'], factory) :
    (global = global || self, factory((global.swagger = global.swagger || {}, global.swagger['typescript-on-time-app-api'] = {}), global.ng.core, global.ng.common.http));
}(this, (function (exports, core, http) { 'use strict';

    /*! *****************************************************************************
    Copyright (c) Microsoft Corporation. All rights reserved.
    Licensed under the Apache License, Version 2.0 (the "License"); you may not use
    this file except in compliance with the License. You may obtain a copy of the
    License at http://www.apache.org/licenses/LICENSE-2.0

    THIS CODE IS PROVIDED ON AN *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
    KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
    WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
    MERCHANTABLITY OR NON-INFRINGEMENT.

    See the Apache Version 2.0 License for specific language governing permissions
    and limitations under the License.
    ***************************************************************************** */

    function __decorate(decorators, target, key, desc) {
        var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
        if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
        else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
        return c > 3 && r && Object.defineProperty(target, key, r), r;
    }

    function __param(paramIndex, decorator) {
        return function (target, key) { decorator(target, key, paramIndex); }
    }

    function __metadata(metadataKey, metadataValue) {
        if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(metadataKey, metadataValue);
    }

    function __values(o) {
        var m = typeof Symbol === "function" && o[Symbol.iterator], i = 0;
        if (m) return m.call(o);
        return {
            next: function () {
                if (o && i >= o.length) o = void 0;
                return { value: o && o[i++], done: !o };
            }
        };
    }

    var BASE_PATH = new core.InjectionToken('basePath');
    var COLLECTION_FORMATS = {
        'csv': ',',
        'tsv': '   ',
        'ssv': ' ',
        'pipes': '|'
    };

    var Configuration = /** @class */ (function () {
        function Configuration(configurationParameters) {
            if (configurationParameters === void 0) { configurationParameters = {}; }
            this.apiKeys = configurationParameters.apiKeys;
            this.username = configurationParameters.username;
            this.password = configurationParameters.password;
            this.accessToken = configurationParameters.accessToken;
            this.basePath = configurationParameters.basePath;
            this.withCredentials = configurationParameters.withCredentials;
        }
        /**
         * Select the correct content-type to use for a request.
         * Uses {@link Configuration#isJsonMime} to determine the correct content-type.
         * If no content type is found return the first found type if the contentTypes is not empty
         * @param contentTypes - the array of content types that are available for selection
         * @returns the selected content-type or <code>undefined</code> if no selection could be made.
         */
        Configuration.prototype.selectHeaderContentType = function (contentTypes) {
            var _this = this;
            if (contentTypes.length == 0) {
                return undefined;
            }
            var type = contentTypes.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return contentTypes[0];
            }
            return type;
        };
        /**
         * Select the correct accept content-type to use for a request.
         * Uses {@link Configuration#isJsonMime} to determine the correct accept content-type.
         * If no content type is found return the first found type if the contentTypes is not empty
         * @param accepts - the array of content types that are available for selection.
         * @returns the selected content-type or <code>undefined</code> if no selection could be made.
         */
        Configuration.prototype.selectHeaderAccept = function (accepts) {
            var _this = this;
            if (accepts.length == 0) {
                return undefined;
            }
            var type = accepts.find(function (x) { return _this.isJsonMime(x); });
            if (type === undefined) {
                return accepts[0];
            }
            return type;
        };
        /**
         * Check if the given MIME is a JSON MIME.
         * JSON MIME examples:
         *   application/json
         *   application/json; charset=UTF8
         *   APPLICATION/JSON
         *   application/vnd.company+json
         * @param mime - MIME (Multipurpose Internet Mail Extensions)
         * @return True if the given MIME is JSON, false otherwise.
         */
        Configuration.prototype.isJsonMime = function (mime) {
            var jsonMime = new RegExp('^(application\/json|[^;/ \t]+\/[^;/ \t]+[+]json)[ \t]*(;.*)?$', 'i');
            return mime != null && (jsonMime.test(mime) || mime.toLowerCase() === 'application/json-patch+json');
        };
        return Configuration;
    }());

    /**
     * OnTimeApp API
     * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
     *
     * OpenAPI spec version: 1
     * Contact: sergiojj932@gmail.com
     *
     * NOTE: This class is auto generated by the swagger code generator program.
     * https://github.com/swagger-api/swagger-codegen.git
     * Do not edit the class manually.
     */
    var CheckInRecordsService = /** @class */ (function () {
        function CheckInRecordsService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = 'https://localhost';
            this.defaultHeaders = new http.HttpHeaders();
            this.configuration = new Configuration();
            if (basePath) {
                this.basePath = basePath;
            }
            if (configuration) {
                this.configuration = configuration;
                this.basePath = basePath || configuration.basePath || this.basePath;
            }
        }
        /**
         * @param consumes string[] mime-types
         * @return true: consumes contains 'multipart/form-data', false: otherwise
         */
        CheckInRecordsService.prototype.canConsumeForm = function (consumes) {
            var form = 'multipart/form-data';
            try {
                for (var consumes_1 = __values(consumes), consumes_1_1 = consumes_1.next(); !consumes_1_1.done; consumes_1_1 = consumes_1.next()) {
                    var consume = consumes_1_1.value;
                    if (form === consume) {
                        return true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (consumes_1_1 && !consumes_1_1.done && (_a = consumes_1.return)) _a.call(consumes_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            return false;
            var e_1, _a;
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsEmailDateGet = function (email, date, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1CheckInRecordsEmailDateGet.');
            }
            if (date === null || date === undefined) {
                throw new Error('Required parameter date was null or undefined when calling apiV1CheckInRecordsEmailDateGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/CheckInRecords/" + encodeURIComponent(String(email)) + "/" + encodeURIComponent(String(date.toISOString())), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsEmailFromToGet = function (email, from, to, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1CheckInRecordsEmailFromToGet.');
            }
            if (from === null || from === undefined) {
                throw new Error('Required parameter from was null or undefined when calling apiV1CheckInRecordsEmailFromToGet.');
            }
            if (to === null || to === undefined) {
                throw new Error('Required parameter to was null or undefined when calling apiV1CheckInRecordsEmailFromToGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/CheckInRecords/" + encodeURIComponent(String(email)) + "/" + encodeURIComponent(String(from.toISOString())) + "/" + encodeURIComponent(String(to.toISOString())), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsEmailGet = function (email, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1CheckInRecordsEmailGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/CheckInRecords/" + encodeURIComponent(String(email)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsGet = function (observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/CheckInRecords", {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsRegisterPost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/CheckInRecords/Register", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService.prototype.apiV1CheckInRecordsWorkedTimeEmailDateGet = function (email, date, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1CheckInRecordsWorkedTimeEmailDateGet.');
            }
            if (date === null || date === undefined) {
                throw new Error('Required parameter date was null or undefined when calling apiV1CheckInRecordsWorkedTimeEmailDateGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/CheckInRecords/WorkedTime/" + encodeURIComponent(String(email)) + "/" + encodeURIComponent(String(date.toISOString())), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        CheckInRecordsService = __decorate([
            core.Injectable(),
            __param(1, core.Optional()), __param(1, core.Inject(BASE_PATH)), __param(2, core.Optional()),
            __metadata("design:paramtypes", [http.HttpClient, String, Configuration])
        ], CheckInRecordsService);
        return CheckInRecordsService;
    }());

    /**
     * OnTimeApp API
     * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
     *
     * OpenAPI spec version: 1
     * Contact: sergiojj932@gmail.com
     *
     * NOTE: This class is auto generated by the swagger code generator program.
     * https://github.com/swagger-api/swagger-codegen.git
     * Do not edit the class manually.
     */
    var HolidaysService = /** @class */ (function () {
        function HolidaysService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = 'https://localhost';
            this.defaultHeaders = new http.HttpHeaders();
            this.configuration = new Configuration();
            if (basePath) {
                this.basePath = basePath;
            }
            if (configuration) {
                this.configuration = configuration;
                this.basePath = basePath || configuration.basePath || this.basePath;
            }
        }
        /**
         * @param consumes string[] mime-types
         * @return true: consumes contains 'multipart/form-data', false: otherwise
         */
        HolidaysService.prototype.canConsumeForm = function (consumes) {
            var form = 'multipart/form-data';
            try {
                for (var consumes_1 = __values(consumes), consumes_1_1 = consumes_1.next(); !consumes_1_1.done; consumes_1_1 = consumes_1.next()) {
                    var consume = consumes_1_1.value;
                    if (form === consume) {
                        return true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (consumes_1_1 && !consumes_1_1.done && (_a = consumes_1.return)) _a.call(consumes_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            return false;
            var e_1, _a;
        };
        HolidaysService.prototype.apiV1HolidaysApprovePost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Holidays/Approve", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        HolidaysService.prototype.apiV1HolidaysDeletePost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Holidays/Delete", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        HolidaysService.prototype.apiV1HolidaysEmailFromToGet = function (email, from, to, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1HolidaysEmailFromToGet.');
            }
            if (from === null || from === undefined) {
                throw new Error('Required parameter from was null or undefined when calling apiV1HolidaysEmailFromToGet.');
            }
            if (to === null || to === undefined) {
                throw new Error('Required parameter to was null or undefined when calling apiV1HolidaysEmailFromToGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Holidays/" + encodeURIComponent(String(email)) + "/" + encodeURIComponent(String(from)) + "/" + encodeURIComponent(String(to)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        HolidaysService.prototype.apiV1HolidaysRegisterPost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Holidays/Register", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        HolidaysService.prototype.apiV1HolidaysToApproveEmailGet = function (email, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1HolidaysToApproveEmailGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Holidays/ToApprove/" + encodeURIComponent(String(email)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        HolidaysService = __decorate([
            core.Injectable(),
            __param(1, core.Optional()), __param(1, core.Inject(BASE_PATH)), __param(2, core.Optional()),
            __metadata("design:paramtypes", [http.HttpClient, String, Configuration])
        ], HolidaysService);
        return HolidaysService;
    }());

    /**
     * OnTimeApp API
     * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
     *
     * OpenAPI spec version: 1
     * Contact: sergiojj932@gmail.com
     *
     * NOTE: This class is auto generated by the swagger code generator program.
     * https://github.com/swagger-api/swagger-codegen.git
     * Do not edit the class manually.
     */
    var IdentityService = /** @class */ (function () {
        function IdentityService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = 'https://localhost';
            this.defaultHeaders = new http.HttpHeaders();
            this.configuration = new Configuration();
            if (basePath) {
                this.basePath = basePath;
            }
            if (configuration) {
                this.configuration = configuration;
                this.basePath = basePath || configuration.basePath || this.basePath;
            }
        }
        /**
         * @param consumes string[] mime-types
         * @return true: consumes contains 'multipart/form-data', false: otherwise
         */
        IdentityService.prototype.canConsumeForm = function (consumes) {
            var form = 'multipart/form-data';
            try {
                for (var consumes_1 = __values(consumes), consumes_1_1 = consumes_1.next(); !consumes_1_1.done; consumes_1_1 = consumes_1.next()) {
                    var consume = consumes_1_1.value;
                    if (form === consume) {
                        return true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (consumes_1_1 && !consumes_1_1.done && (_a = consumes_1.return)) _a.call(consumes_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            return false;
            var e_1, _a;
        };
        IdentityService.prototype.apiV1IdentityInitPost = function (observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Identity/init", null, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        IdentityService.prototype.apiV1IdentityLoginPost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Identity/login", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        IdentityService.prototype.apiV1IdentityRegisterPost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Identity/register", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        IdentityService = __decorate([
            core.Injectable(),
            __param(1, core.Optional()), __param(1, core.Inject(BASE_PATH)), __param(2, core.Optional()),
            __metadata("design:paramtypes", [http.HttpClient, String, Configuration])
        ], IdentityService);
        return IdentityService;
    }());

    /**
     * OnTimeApp API
     * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
     *
     * OpenAPI spec version: 1
     * Contact: sergiojj932@gmail.com
     *
     * NOTE: This class is auto generated by the swagger code generator program.
     * https://github.com/swagger-api/swagger-codegen.git
     * Do not edit the class manually.
     */
    var RolesService = /** @class */ (function () {
        function RolesService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = 'https://localhost';
            this.defaultHeaders = new http.HttpHeaders();
            this.configuration = new Configuration();
            if (basePath) {
                this.basePath = basePath;
            }
            if (configuration) {
                this.configuration = configuration;
                this.basePath = basePath || configuration.basePath || this.basePath;
            }
        }
        /**
         * @param consumes string[] mime-types
         * @return true: consumes contains 'multipart/form-data', false: otherwise
         */
        RolesService.prototype.canConsumeForm = function (consumes) {
            var form = 'multipart/form-data';
            try {
                for (var consumes_1 = __values(consumes), consumes_1_1 = consumes_1.next(); !consumes_1_1.done; consumes_1_1 = consumes_1.next()) {
                    var consume = consumes_1_1.value;
                    if (form === consume) {
                        return true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (consumes_1_1 && !consumes_1_1.done && (_a = consumes_1.return)) _a.call(consumes_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            return false;
            var e_1, _a;
        };
        RolesService.prototype.apiV1RolesGet = function (observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Roles", {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RolesService.prototype.apiV1RolesRegisterPost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Roles/register", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        RolesService = __decorate([
            core.Injectable(),
            __param(1, core.Optional()), __param(1, core.Inject(BASE_PATH)), __param(2, core.Optional()),
            __metadata("design:paramtypes", [http.HttpClient, String, Configuration])
        ], RolesService);
        return RolesService;
    }());

    /**
     * OnTimeApp API
     * No description provided (generated by Swagger Codegen https://github.com/swagger-api/swagger-codegen)
     *
     * OpenAPI spec version: 1
     * Contact: sergiojj932@gmail.com
     *
     * NOTE: This class is auto generated by the swagger code generator program.
     * https://github.com/swagger-api/swagger-codegen.git
     * Do not edit the class manually.
     */
    var UsersService = /** @class */ (function () {
        function UsersService(httpClient, basePath, configuration) {
            this.httpClient = httpClient;
            this.basePath = 'https://localhost';
            this.defaultHeaders = new http.HttpHeaders();
            this.configuration = new Configuration();
            if (basePath) {
                this.basePath = basePath;
            }
            if (configuration) {
                this.configuration = configuration;
                this.basePath = basePath || configuration.basePath || this.basePath;
            }
        }
        /**
         * @param consumes string[] mime-types
         * @return true: consumes contains 'multipart/form-data', false: otherwise
         */
        UsersService.prototype.canConsumeForm = function (consumes) {
            var form = 'multipart/form-data';
            try {
                for (var consumes_1 = __values(consumes), consumes_1_1 = consumes_1.next(); !consumes_1_1.done; consumes_1_1 = consumes_1.next()) {
                    var consume = consumes_1_1.value;
                    if (form === consume) {
                        return true;
                    }
                }
            }
            catch (e_1_1) { e_1 = { error: e_1_1 }; }
            finally {
                try {
                    if (consumes_1_1 && !consumes_1_1.done && (_a = consumes_1.return)) _a.call(consumes_1);
                }
                finally { if (e_1) throw e_1.error; }
            }
            return false;
            var e_1, _a;
        };
        UsersService.prototype.apiV1UsersAddRolePost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Users/AddRole", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersAddSubordiantePost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Users/AddSubordiante", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersEmailGet = function (email, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1UsersEmailGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Users/" + encodeURIComponent(String(email)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersGet = function (observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Users", {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersGetRolesEmailGet = function (email, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1UsersGetRolesEmailGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Users/GetRoles/" + encodeURIComponent(String(email)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersRemoveRolePost = function (body, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            // to determine the Content-Type header
            var consumes = [
                'application/json-patch+json',
                'application/json',
                'text/json',
                'application/_*+json'
            ];
            var httpContentTypeSelected = this.configuration.selectHeaderContentType(consumes);
            if (httpContentTypeSelected != undefined) {
                headers = headers.set('Content-Type', httpContentTypeSelected);
            }
            return this.httpClient.post(this.basePath + "/api/v1/Users/RemoveRole", body, {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService.prototype.apiV1UsersSubordinatesEmailGet = function (email, observe, reportProgress) {
            if (observe === void 0) { observe = 'body'; }
            if (reportProgress === void 0) { reportProgress = false; }
            if (email === null || email === undefined) {
                throw new Error('Required parameter email was null or undefined when calling apiV1UsersSubordinatesEmailGet.');
            }
            var headers = this.defaultHeaders;
            // authentication (Bearer) required
            if (this.configuration.apiKeys["Authorization"]) {
                headers = headers.set('Authorization', this.configuration.apiKeys["Authorization"]);
            }
            // to determine the Accept header
            var httpHeaderAccepts = [
                'text/plain',
                'application/json',
                'text/json'
            ];
            var httpHeaderAcceptSelected = this.configuration.selectHeaderAccept(httpHeaderAccepts);
            if (httpHeaderAcceptSelected != undefined) {
                headers = headers.set('Accept', httpHeaderAcceptSelected);
            }
            return this.httpClient.get(this.basePath + "/api/v1/Users/Subordinates/" + encodeURIComponent(String(email)), {
                withCredentials: this.configuration.withCredentials,
                headers: headers,
                observe: observe,
                reportProgress: reportProgress
            });
        };
        UsersService = __decorate([
            core.Injectable(),
            __param(1, core.Optional()), __param(1, core.Inject(BASE_PATH)), __param(2, core.Optional()),
            __metadata("design:paramtypes", [http.HttpClient, String, Configuration])
        ], UsersService);
        return UsersService;
    }());

    var APIS = [CheckInRecordsService, HolidaysService, IdentityService, RolesService, UsersService];

    var ApiModule = /** @class */ (function () {
        function ApiModule(parentModule, http) {
            if (parentModule) {
                throw new Error('ApiModule is already loaded. Import in your base AppModule only.');
            }
            if (!http) {
                throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                    'See also https://github.com/angular/angular/issues/20575');
            }
        }
        ApiModule_1 = ApiModule;
        ApiModule.forRoot = function (configurationFactory) {
            return {
                ngModule: ApiModule_1,
                providers: [{ provide: Configuration, useFactory: configurationFactory }]
            };
        };
        ApiModule = ApiModule_1 = __decorate([
            core.NgModule({
                imports: [],
                declarations: [],
                exports: [],
                providers: [
                    CheckInRecordsService,
                    HolidaysService,
                    IdentityService,
                    RolesService,
                    UsersService
                ]
            }),
            __param(0, core.Optional()), __param(0, core.SkipSelf()),
            __param(1, core.Optional()),
            __metadata("design:paramtypes", [ApiModule,
                http.HttpClient])
        ], ApiModule);
        return ApiModule;
        var ApiModule_1;
    }());

    exports.APIS = APIS;
    exports.ApiModule = ApiModule;
    exports.BASE_PATH = BASE_PATH;
    exports.COLLECTION_FORMATS = COLLECTION_FORMATS;
    exports.CheckInRecordsService = CheckInRecordsService;
    exports.Configuration = Configuration;
    exports.HolidaysService = HolidaysService;
    exports.IdentityService = IdentityService;
    exports.RolesService = RolesService;
    exports.UsersService = UsersService;

    Object.defineProperty(exports, '__esModule', { value: true });

})));
