// @swagger/typescript-on-time-app-api/esm5/api.module.js
import * as tslib_1 from "tslib";
import { NgModule, SkipSelf, Optional } from '@angular/core';
import { Configuration } from './configuration';
import { HttpClient } from '@angular/common/http';
import { CheckInRecordsService } from './api/checkInRecords.service';
import { HolidaysService } from './api/holidays.service';
import { IdentityService } from './api/identity.service';
import { RolesService } from './api/roles.service';
import { UsersService } from './api/users.service';
var ApiModule = /** @class */ (function () {
    function ApiModule(parentModule, http) {
        if (parentModule) {
            throw new Error('ApiModule is already loaded. Import in your base AppModule only.');
        }
        if (!http) {
            throw new Error('You need to import the HttpClientModule in your AppModule! \n' +
                'See also https://github.com/angular/angular/issues/20575');
        }
    }
    ApiModule_1 = ApiModule;
    ApiModule.forRoot = function (configurationFactory) {
        return {
            ngModule: ApiModule_1,
            providers: [{ provide: Configuration, useFactory: configurationFactory }]
        };
    };
    ApiModule = ApiModule_1 = tslib_1.__decorate([
        NgModule({
            imports: [],
            declarations: [],
            exports: [],
            providers: [
                CheckInRecordsService,
                HolidaysService,
                IdentityService,
                RolesService,
                UsersService
            ]
        }),
        tslib_1.__param(0, Optional()), tslib_1.__param(0, SkipSelf()),
        tslib_1.__param(1, Optional()),
        tslib_1.__metadata("design:paramtypes", [ApiModule,
            HttpClient])
    ], ApiModule);
    return ApiModule;
    var ApiModule_1;
}());
export { ApiModule };
